MSAL/src/instance/MSALAdfsAuthorityResolver.m

```diff
@@ -133,15 +131,25 @@ - (void)queryEnrollmentServerEndpointForDomain:(NSString *)domain
 {
     CHECK_ERROR_COMPLETION(domain, context, MSALErrorInvalidParameter, @"Domain cannot be nil.");
     
-    MSALWebAuthRequest *request =
-    [[MSALWebAuthRequest alloc] initWithURL:[MSALAdfsAuthorityResolver urlForDrsDiscoveryForDomain:domain adfsType:type]
-                                    context:context];
-    
-    [request sendGet:^(MSALHttpResponse *response, NSError *error) {
+    NSURL *url = [MSALAdfsAuthorityResolver urlForDrsDiscoveryForDomain:domain adfsType:type];
+    MSIDAADAuthorityValidationRequest *request = [[MSIDAADAuthorityValidationRequest alloc] initWithUrl:url
+                                                                                                context:context];
+    [request sendWithBlock:^(id response, NSError *error) {
+        
+        [request finishAndInvalidate];
+        
         CHECK_COMPLETION(!error);
         
+        if(response && ![response isKindOfClass:[NSMutableDictionary class]])
+        {
+            NSError *localError = CREATE_MSID_LOG_ERROR(context, MSALErrorInternal, @"response is not of the expected type: NSMutableDictionary.");
+            completionBlock(nil, localError);
```

Comment:
nit: can we check that completionBlock is not nil?

Reply:
done